Add troubleshooting to build_p4_fusion.md (#48763)
sourcegraph/sourcegraph: doc/dev/background-information/build_p4_fusion.md

1. Run `nix build ".#p4-fusion" --verbose --extra-experimental-features nix-command --extra-experimental-features flakes`

If the build completes successfully you should have a `p4-fusion` binary in `./result/bin/p4-fusion` which you can copy somewhere in your `$PATH`

## Troubleshooting

### `nix build` fails with "hash mismatch" error referencing `helix-core-api.drv`

The `p4-fusion` dependencies specified in [the `srcs` array of `dev/nix/p4-fusion.nix`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@8d513301f8a12f7c7e0b5a66ed20ba14b6679cca/-/blob/dev/nix/p4-fusion.nix?L45-77) are sometimes updated without being versioned, so their hashes change, which causes "hash mismatch" errors.

**Example**

```bash
building '/nix/store/cb2ssx8vykl1ghb4k87yp3q6wnvfvjj2-helix-core-api.drv'...
error: hash mismatch in fixed-output derivation '/nix/store/cb2ssx8vykl1ghb4k87yp3q6wnvfvjj2-helix-core-api.drv':
         specified: sha256-8yX9sjE1f798ns0UmHXz5I8kdpUXHS01FG46SU8nsZw=
            got:    sha256-gaYvQOX8nvMIMHENHB0+uklyLcmeXT5gjGGcVC9TTtE=
error: 1 dependencies of derivation '/nix/store/m409z1rq40bwzvvndbnghrrxm000zd9v-p4-fusion.drv' failed to build
```

> To update the hashes, use `nix` tools to download the offending archive, unpack it, calculate the hash of the unpacked archive, and convert the format to an SRI representation that includes the type of the hash.

Reviewer: Im thinking perhaps a rewording along these lines might sound less intimidating. There can be a lot of hesitancy in adopting Nix due to the learning curve, so ideally I'd like to have things seem like less than a major hassle to do 😄 feel free to use any (or none) of this suggestion

```suggestion
To update the hashes, we use `nix` tools to calculate the hashes of the unpacked archives, and convert the format to an SRI representation that includes the type of the hash.
```

> Get the URL of the archive from the part of the `srcs` array that matches the OS and architecture of your local system.

Reviewer: Im assuming that if one is updated that all would be too, so maybe worth nudging the next person to do it for all 🙂

Author: Good idea.

**Example**

Replace `<url of archive>` with the value of the archive url from `srcs`

```bash
hash_type=sha256
hash_value=$(nix-prefetch-url --type "${hash_type}" --unpack <url of archive>)

> nix --extra-experimental-features nix-command hash to-sri "${hash_type}:${hash_value}"

Reviewer: Thinking it might be clearer if each line was prefixed with `$`? I usually see that when theres a (non-inline) code block I believe

Author: Heh, I usually leave out the prompt because it makes copy/paste more difficult. I do want to be consistent with the rest of docsite, though; searching to see if there is a consensus...

Author: Hmmmm, [seems to be a mix](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+file:doc/.*%5C.md+/%60%60%60%28sh%7Cbash%7Czsh%7Cksh%29/&patternType=standard&sm=1&groupBy=path)

Author: I added the prompt here because it does look more clear, but I left it out of the example I added so that it can be copy/pasted
